src/vivarium_testing_utils/automated_validation/data_transformation/calculations.py

```diff
@@ -131,3 +131,27 @@ def get_singular_indices(data: pd.DataFrame) -> dict[str, Any]:
         if data.index.get_level_values(index_level).nunique() == 1:
             singular_metadata[index_level] = data.index.get_level_values(index_level)[0]
     return singular_metadata
+
+
+def weighted_average(
+    data: pd.DataFrame, weights: pd.DataFrame, strata: list[str] | None = None
```

Comment:
Nit: are we sure `strata` is the right word here? We've traditionally used `stratifications` (which may or may not really be a word), but strata seems more like it means the same thing as the categories within a single stratification, which this is not.